xs/Buffer.xs

```diff
@@ -0,0 +1,53 @@
+MODULE = FFI::Platypus PACKAGE = FFI::Platypus::Buffer
+
+# Stolen from Data::Peek::DGrow
+void
+grow (sv, size, ... )
+    SV     *sv
+    IV      size
+ 
+  PROTOTYPE: $$;$
+  PREINIT:
+    int clear = 1;  
+  PPCODE:
+    if ( items > 2 )
+        clear = SvIV(ST(2));
+
+    if (SvROK (sv))
+        sv = SvRV (sv);
+    if (!SvPOK (sv))
+        sv_setpvn (sv, "", 0);
+    /* don't need the contents; avoid copying into the new memory */
+
+    if (clear)
```

Comment:
```suggestion
    else if (clear)
```

No I agree that we should use `SvPVCLEAR`, but I don't think we need to clear in the event that we call `sv_setpvn(sv, "", 0);` here

Reply:
Understood.  The code now looks like this:
```
    /* if not a string turn it into an empty one, or if clearing is
       requested, reset string length */
    if (!SvPOK (sv) || clear ) {
#     if PERL_API_VERSION >= 26
        SvPVCLEAR(sv);
#else
        sv_setpvn (sv, "", 0);
#endif
    }

```

Reply:
style nit, no space between the `#` and `if`:
```c
#if PERL_API_VERSION >= 26
```

This is good so long as it is okay to call `SvPVCLEAR` on `undef`, which I think there is already a test for.

Reply:
I'll add a test for that.

Reply:
Slightly unexpected behavior when passing an undefined value to `set_used_length`, e.g.
```
my $str;
set_used_length( $str, 100 );
```
An undefined SV is converted to a PV via `sv_setpn( sv, "", 0 ),` which allocates a minimum buffer length (on my box 10 bytes).   For safety, `set_used_length ` caps the string length to the length of the PV's buffer.  In this case, that results in something starting out as undefined ending up as a string with length 10.

Not necessarily anything wrong, but unexpected.  My instinct is to force it to a length of 0 in this case.